Configure and install the Cobra binary

mjeronimo opened the pull request:
Signed-off-by: Michael Jeronimo <michael.jeronimo@openrobotics.org>

Commit:
Configure and install the Cobra binary

Signed-off-by: Michael Jeronimo <michael.jeronimo@openrobotics.org>

nuclearsandwich commented on CMakeLists.txt:
```diff
@@ -15,13 +14,17 @@ ExternalProject_Add(cobra-3.6
   GIT_REPOSITORY https://github.com/nimble-code/cobra
   GIT_TAG ab06e676f9123ec07011a09c38daffaa4fbfdc60
-  INSTALL_DIR "${CMAKE_CURRENT_BINARY_DIR}/cobra_install"
+  INSTALL_DIR "${CMAKE_CURRENT_BINARY_DIR}/bin_linux"
```
What change is this intended to achieve? The install target from Cobra's makefiles is not configurable so the install dir here doesn't help us much. I set it by convention but haven't yet made it do anything.

If the intent is to target the bin_linux directory that cobra actually installs to it should probably be
```suggestion
  INSTALL_DIR <SOURCE_DIR>/bin_linux
```

mjeronimo replied:
I had a misunderstanding about INSTALL_DIR. Since it's not actually used in this case to install the binary, I'll revert to match convention.

nuclearsandwich commented on CMakeLists.txt:
```diff
@@ -15,13 +14,17 @@ ExternalProject_Add(cobra-3.6
   ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>/src make linux && ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>/src_app make all
   INSTALL_COMMAND
-  ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>/src make install_linux MAN=/dev/null && ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>/src_app make install_linux
+  ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>/src make install_linux MAN=/dev/null && ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>/src_app make install_linux && <SOURCE_DIR>/bin_linux/cobra -configure <SOURCE_DIR>
```
I don't think that we can run this configure step without affecting the host user's home which we definitely ought to avoid.

Instead of running `cobra -configure` what we'll do instead is install the rules files into `share/cobra/rules` and then using an environment hook to set the `C_BASE` environment variable as described in the [cobra README.md](https://github.com/nimble-code/Cobra/blob/master/README.md).

mjeronimo replied:
Sounds good. I'll update.

mjeronimo replied:
Updated to install the rules to the package share directory and set C_BASE using an environment hook.

nuclearsandwich reviewed (changes requested):
Thanks for helping to keep this rolling. I've got comments on a couple of the changes so once those are addressed we can add these and continue making improvements in followup.

Commit:
Change cobra configuration method

Install the cobra rules files into the package share directory.
Then, instead of invoking "cobra -configure", use and environment
hook to set the C_BASE environment variable to point to the
rules directory.

Signed-off-by: Michael Jeronimo <michael.jeronimo@openrobotics.org>

Commit:
Remove stray file

Signed-off-by: Michael Jeronimo <michael.jeronimo@openrobotics.org>

Commit:
Correct the INSTALL_DIR to match convention

Signed-off-by: Michael Jeronimo <michael.jeronimo@openrobotics.org>
